src/components/PDFView/PDFPasswordForm.js

```diff
@@ -0,0 +1,151 @@
+
+    ...withLocalizePropTypes,
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    isPasswordInvalid: false,
+    shouldAutofocusPasswordField: false,
+    shouldShowLoadingIndicator: false,
+    onAvoidKeyboard: () => {},
+};
+
+class PDFPasswordForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            password: '',
+            validationErrorText: '',
+            isEditingInProgress: true,
+            shouldShowForm: false,
+        };
+        this.submitPassword = this.submitPassword.bind(this);
+        this.validate = this.validate.bind(this);
+        this.validateOnBlur = this.validateOnBlur.bind(this);
+        this.updatePassword = this.updatePassword.bind(this);
+        this.showForm = this.showForm.bind(this);
+    }
+
+    submitPassword() {
+        this.validate();
+        if (!_.isEmpty(this.state.password)) {
+            this.setState({isEditingInProgress: false});
+            this.props.onSubmit(this.state.password);
+        }
+    }
+
+    updatePassword(password) {
+        if (!_.isEmpty(password)) {
+            this.setState({validationErrorText: ''});
+        }
+        this.setState({password, isEditingInProgress: true});
+    }
+
+    validateOnBlur() {
+        // Notify parent that keyboard is no longer visible (for mobile devices).
+        this.props.onAvoidKeyboard(false);
+
+        this.validate();
+    }
+
+    validate() {
+        if (!_.isEmpty(this.state.password)) {
+            return;
+        }
+        this.setState({
+            validationErrorText: this.props.translate('attachmentView.passwordRequired'),
+        });
+    }
+
+    showForm() {
+        this.setState({shouldShowForm: true});
+    }
+
+    render() {
+        // Use container styles appropriate for screen size.
+        const containerStyles = this.props.isSmallScreenWidth
+            ? styles.pdfPasswordForm.narrowScreen
+            : styles.pdfPasswordForm.wideScreen;
+
+        return (
+            <>
+                {this.state.shouldShowForm ? (
+                    <View style={containerStyles}>
+                        <Text style={styles.mb4}>
+                            {this.props.translate('attachmentView.pdfPasswordForm.formLabel')}
+                        </Text>
+                        <TextInput
+                            label={this.props.translate('common.password')}
+                            autoComplete="off"
+                            autoCorrect={false}
+                            textContentType="password"
+                            onChangeText={this.updatePassword}
+                            returnKeyType="done"
+                            onSubmitEditing={this.submitPassword}
+                            errorText={this.state.validationErrorText}
+                            onBlur={this.validateOnBlur}
+                            onFocus={() => this.props.onAvoidKeyboard(true)}
```

Comment:
The `onAvoidKeyboard` handler is used to hide the send button on Android devices when the keyboard is open.

According to the mockups the send button should not be visible when the keyboard is open. On iOS this happens automatically since the keyboard covers the send button. On Android, however, the send button must be explicitly hidden.

Reply:
Ok, I am just worried about this deeply nested prop. Can we listen for Keyboard events on the AttachmentModal itself and then hide the button using that? On web,  there are a few ways to do this. 
1. You can listen for visualViewport events. 
2. Or we can use `focusin/focusout` event to track keyboard opening and closing

Reply:
Why does the nestet prop worry you?

And what are `focusin/focusout` events?

Reply:
Let's hold this until we get the UI confirmed.

Reply:
Let's leave this change for now.